### DISCUSSION TITLE
Do All Themes needs a Layout shape?

### DISCUSSION THREAD

**Discussion by FireEagle97** [OrchardCMS/OrchardCore#15128] [Q&A]
I'm trying to create a theme that uses MVC instead of the usual liquid but I'm having an issue in rendering _Layout.cshtml.
Here is what I tried:
-Set the path in _ViewStart as
 `@{
    Layout = "_Layout";
}
`
-Set the path in _ViewStart as
`
@{
    Layout = "~/Views/Shared/_Layout.cshtml";
}
`
it gives me Exception: Shape type 'Layout' not found
My question is Do I need to use shapes and define a Layout shape even if I'm using MVC?And is there a difference in setting up the theme whether in liquid, razor or MVC
![err](https://github.com/OrchardCMS/OrchardCore/assets/62710297/bab7dcb3-edf8-4083-8bf6-8ba2b87e4425)

**Comment by Piedone** [comment]
Just to clarify, by MVC do you mean using Razor templates? Because if you create a theme with Razor template, then yes, you still need a Layout shape. You need to put it into the root of the `Views` folder, like [TheTheme does](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Themes/TheTheme/Views/Layout.cshtml).

If you mean to create your own MVC frontend without Orchard Core theming, then you're after [the decoupled approach](https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/).

**Reply by FireEagle97** [reply]
What's the advantage of using the Orchard Core Theming versus using my own Frontend?

**Reply by Piedone** [reply]
When you use your own frontend, you build the whole frontend from scratch. If you create a theme, you can make that extend an existing one (BaseTheme) and features like content items have a default display out of the box.

**Comment by ns8482e** [comment]
By defining `@{ Layout = "_Layout"; }` you are opting out for not using OC theming

OC theming layout are not same as MVC layout, OC Layout are Shapes.

to use OC theming and Layout shapes create `Layout.cshtml` in `Views` folder and do not assign any value to Layout property on page.

Once the theme is enabled, all pages will use your Layout.cshtml shape

For some reason if you need to specify specific layout shape to specific page, you can define `Views\OtherLayout.cshtml` and you can define `ViewLayout = "OtherLayout" ` on that page
